clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -664,6 +673,16 @@ raw2trace_t::process_offline_entry(raw2trace_thread_data_t *tdata,
         DR_CHECK(reinterpret_cast<trace_entry_t *>(buf) == buf_base,
                  "We shouldn't have buffered anything before calling "
                  "append_bb_entries");
+        if (tdata->saw_filter_endpoint_) {
```

Comment:
Something looks off here: it looks like after we see the ENDPOINT marker we'll enter this if() over and over for every record in the buffer.

The code here could be moved into the if() that actually sees the marker, right?  And we could get rid of "saw_filter_endpoint_" which by its name sounds like it means "ever seen by this thead" yet really it means "seen in the current buffer" b/c it's cleared per buffer; yet that clearing is not enough.  If some kind of separate variable is really needed (it doesn't look like it to me though) please rename to make its scope clear, and avoid entering here over and over.

Reply:
This variable was not needed. I have removed it.